gst/nnstreamer/tensor_repo.c

```diff
@@ -86,17 +175,30 @@ gst_tensor_repo_add_repodata (guint nth)
  * @brief push GstBuffer into repo
  */
 gboolean
-gst_tensor_repo_set_buffer (guint nth, GstBuffer * buffer)
+gst_tensor_repo_set_buffer (guint nth, GstBuffer * buffer, GstCaps * caps)
 {
   GST_TENSOR_REPO_LOCK (nth);
 
   GstTensorRepoData *data = gst_tensor_repo_get_repodata (nth);
 
+  if (data->eos) {
+    GST_TENSOR_REPO_UNLOCK (nth);
+    return FALSE;
+  }
+
   while (data->buffer != NULL) {
     GST_TENSOR_REPO_WAIT_PULL (nth);
   }
 
-  data->buffer = buffer;
+  data->buffer = gst_buffer_copy (buffer);
```

Comment:
gst_buffer_copy() copies only the data in buffer.
line 195 GstMetaRepo *meta = GST_META_REPO_GET (data->buffer); meta maybe always null.

Reply:
Right. I will delete if(!meta). Thanks.